app/src/components/misc/PersonaSurvey/utils.ts

```diff
@@ -22,3 +24,19 @@ export const shuffleOptions = (options: Option[]) => {
   }
   return options;
 };
+
+export const syncUserPersona = async (
+  uid: string,
+  dispatch: any,
+  userPersona: UserPersona
+) => {
+  const database = getDatabase();
+  const userProfileRef = ref(database, `users/${uid}/profile`);
+  const profile: any = (await get(child(userProfileRef, "/"))).toJSON();
+  if (!profile) {
+    return;
+  }
+  if (profile.userPersona)
+    dispatch(actions.syncUserPersonaData(profile.userPersona));
+  else update(userProfileRef, { userPersona });
+};
```

Comment:
This function has redundant code from `app/src/actions/FirebaseActions.js` like `getNodeRef`, `getUserProfilePath`, `getValue`, `updateValue` etc,
Try using those

And the dispatch and update, since they are part of business logic for auth only, can be moved to the `AuthForm` component. Should be easy to do with the above mentioned functions.

Trying to avoid sending `dispatch` as argument to a utility function